storm/src/main/storm/mesos/util/MesosCommon.java

```diff
@@ -125,6 +139,24 @@ public static int portFromTaskId(String taskId) {
     return port;
   }
 
+  public static Map<String, OfferResources> getConsolidatedOfferResourcesPerNode(RotatingMap<Protos.OfferID, Protos.Offer> offers) {
+    Map<String, OfferResources> offerResourcesPerNode = new HashMap<>();
+
+    for (Protos.Offer offer : offers.values()) {
+      String hostName = offer.getHostname();
+
+      OfferResources offerResourcesForHost = offerResourcesPerNode.get(hostName);
+      if (offerResourcesForHost == null) {
+        offerResourcesForHost = new OfferResources(offer);
+        offerResourcesPerNode.put(hostName, offerResourcesForHost);
+      } else {
+        offerResourcesForHost.add(offer);
+      }
+      LOG.info("Available resources at " + hostName + ": " + offerResourcesForHost.toString());
```

Comment:
Maybe do a 2nd loop after this one that prints the complete map?  i.e., it *might* be slightly confusing when 1 host has 2+ offers.  Say there are 2 offers for host `foo.bar`, then there would be 2 log lines of `Available resources at foo.bar: ...` where the resources being printed are aggregated in the 2nd log line.